bali/events/handler.py

```diff
@@ -44,21 +44,26 @@ def register_callback(event_type, callback):
                 'Can not find key:%s at AMQP_CONFIGS' % amqp_config_key
             )
         routing_key = None
-        exchange_type = amqp_config.get('EXCHANGE_TYPE')
+        exchange_type = amqp_config.get('EXCHANGE_TYPE') or 'direct'
         if exchange_type != 'fanout':
             routing_key = amqp_config.get(
                 'ROUTING_KEY'
             ) or _settings.BALI_ROUTING_KEY.format(event_type)
+        # Fails when exchange type is different and exchange name is same
         exchange = Exchange(
-            amqp_config.get('EXCHANGE_NAME', _settings.BALI_EXCHANGE),
+            amqp_config.get(
+                'EXCHANGE_NAME', f'{_settings.BALI_EXCHANGE}_{exchange_type}'
+            ),
             type=exchange_type
         )
-        queue = Queue(
-            amqp_config.get('QUEUE_NAME') or
-            _settings.BALI_QUEUE.format(event_type),
-            exchange=exchange,
-            routing_key=routing_key
-        )
+        queue_name = amqp_config.get('QUEUE_NAME')
+        if not queue_name:
+            if exchange_type == 'fanout':
+                queue_name = _settings.BALI_QUEUE.format('ms')
```

Comment:
Default queue name '{service}.events',  service title is stored in Applicaion.title 

eg:
'product.events' for product services.